plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/expression/RewriteVarcharConstant.java

```diff
@@ -46,6 +47,13 @@ public Optional<String> rewrite(Constant constant, Captures captures, RewriteCon
         if (slice == null) {
             return Optional.empty();
         }
+
+        boolean isAscii = CharMatcher.ascii().matchesAllOf(slice.toStringUtf8());
```

Comment:
We are constructing String from Slice twice here. Let's do it once instead.

Also we should invert the logic - let's test if there is at least single non ascii character.

Reply:
fixed